turtlebot2_drivers/CMakeLists.txt

```diff
@@ -12,28 +12,34 @@ find_package(ament_cmake REQUIRED)
 find_package(geometry_msgs REQUIRED)
 find_package(nav_msgs REQUIRED)
 find_package(rclcpp REQUIRED)
-find_package(rmw_implementation_cmake REQUIRED)
-
-include_directories(${kobuki_driver_INCLUDE_DIRS})
-
-macro(targets)
-  if(NOT target_suffix STREQUAL "")
-    get_rclcpp_information("${rmw_implementation}" "rclcpp${target_suffix}")
-  endif()
-  add_executable(kobuki_node${target_suffix} src/kobuki_node.cpp)
-  ament_target_dependencies(kobuki_node${target_suffix}
-    "rclcpp${target_suffix}"
-    "nav_msgs"
-    "geometry_msgs")
-  target_link_libraries(kobuki_node${target_suffix} ${kobuki_driver_LIBRARIES})
-  add_executable(dumb_teleop${target_suffix} src/dumb_teleop.cpp)
-  ament_target_dependencies(dumb_teleop${target_suffix}
-    "rclcpp${target_suffix}"
-    "nav_msgs"
-    "geometry_msgs")
-
-  install(TARGETS kobuki_node${target_suffix} dumb_teleop${target_suffix}
-          DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
-endmacro()
-
-call_for_each_rmw_implementation(targets GENERATE_DEFAULT)
+find_package(sensor_msgs REQUIRED)
+find_package(tf2_ros REQUIRED)
+
+add_executable(kobuki_node src/kobuki_node.cpp)
+ament_target_dependencies(kobuki_node
+  "geometry_msgs"
+  "nav_msgs"
+  "rclcpp"
+  "sensor_msgs"
+  "tf2_ros")
+# Note that this setting must come after the call to
+# ament_target_dependencies() to avoid including a ROS1 tf2 installation that
+# might be on the system.
+target_include_directories(kobuki_node PRIVATE ${kobuki_driver_INCLUDE_DIRS})
```

Comment:
Good question on linking. Looking at the code, and based on some verification that I just did: in this case we're not getting lucky. For a start, nobody is trying to link against `-ltf2_ros`. The call to `ament_target_dependencies()` is generating an absolute path to the ROS2 lib, which isn't ambiguous. And `${kobuki_driver_LIBRARIES}` doesn't include `tf2_ros`. If that last condition weren't true, we'd have trouble. But I'd argue that that would be an unsupported use case, because you're trying to link the ROS2 tf2 lib into an application that also requires a ROS1 lib that internally is using the ROS1 tf2 lib.

The broader point about being careful in these situations still stands, but I'm not sure that changing package names is the right way forward. Let's discuss that at some point.